tests/unit/Rules/UuidTest.php

```diff
@@ -13,45 +13,182 @@
 
 namespace Respect\Validation\Rules;
 
+use Respect\Validation\Exceptions\ComponentException;
 use Respect\Validation\Test\RuleTestCase;
 
 /**
+ * This class provides tests for the UUID validation.
+ *
  * @group  rule
  * @covers \Respect\Validation\Rules\Uuid
+ *
+ * @author Henrique Moody <henriquemoody@gmail.com>
+ * @author Michael Weimann <mail@michael-weimann.eu>
  */
 class UuidTest extends RuleTestCase
 {
+    /**
+     * Provides valid Uuid version params.
+     *
+     * @return array
+     */
+    public function provideValidUuidConstructorParams(): array
+    {
+        return [
+            [Uuid::VERSION_ALL],
+            [Uuid::VERSION_1],
+            [Uuid::VERSION_2],
+            [Uuid::VERSION_3],
+            [Uuid::VERSION_4],
+            [Uuid::VERSION_5],
+        ];
+    }
+
+    /**
+     * Tests that the Uuid validation can be instantiated with correct params.
+     *
+     * @test
+     * @dataProvider provideValidUuidConstructorParams
+     *
+     * @param mixed $version The version passed to the constructor
+     */
+    public function instantiate($version): void
+    {
+        self::expectNotToPerformAssertions();
+        new Uuid($version);
```

Comment:
I don't think that needs to be tested once you're using that in the data providers.